proxyd/methods.go

```diff
@@ -99,7 +113,7 @@ func (e *EthGetBlockByNumberMethodHandler) PutRPCMethod(ctx context.Context, req
 		if err != nil {
 			return err
 		}
-		if curBlock <= blockNum+uint64(e.numBlockConfirmations) {
+		if curBlock < blockNum+uint64(e.numBlockConfirmations) {
```

Comment:
this is a breaking change but only way its going to work... but its more consistent with how eth_blockNumber caching works...